authorization/invitation/service/invitation_service_blackbox_test.go

```diff
@@ -34,619 +38,693 @@ func (s *invitationServiceBlackBoxTest) SetupTest() {
 	s.invitationRepo = invitationrepo.NewInvitationRepository(s.DB)
 	s.identityRepo = account.NewIdentityRepository(s.DB)
 	s.orgService = s.Application.OrganizationService()
-	s.Application = gormapplication.NewGormDB(s.DB, s.Configuration, factory.WithWITService(&test.DevWITService{}))
+	s.notificationServiceMock = testservice.NewNotificationServiceMock(s.T())
+	s.Application = gormapplication.NewGormDB(s.DB, s.Configuration, factory.WithWITService(&test.DevWITService{}), factory.WithNotificationService(s.notificationServiceMock))
 }
 
-func (s *invitationServiceBlackBoxTest) TestIssueInvitationByIdentityID() {
-	g := s.NewTestGraph()
+func (s *invitationServiceBlackBoxTest) TestIssueInvitation() {
+	s.T().Run("should issue invitation by identity id", func(t *testing.T) {
 
-	// Create a test user - this will be the team admin
-	teamAdmin := g.CreateUser()
+		g := s.NewTestGraph()
 
-	// Create a team
-	team := g.CreateTeam()
+		// Create a test user - this will be the team admin
+		teamAdmin := g.CreateUser()
 
-	// Create an admin role and assign it to the team admin
-	r := g.CreateRole(g.LoadResourceType(authorization.IdentityResourceTypeTeam))
-	r.AddScope(authorization.ManageTeamMembersScope)
+		// Create a team
+		team := g.CreateTeam()
 
-	team.AssignRole(teamAdmin.Identity(), r.Role())
+		// Create an admin role and assign it to the team admin
+		r := g.CreateRole(g.LoadResourceType(authorization.IdentityResourceTypeTeam))
+		r.AddScope(authorization.ManageTeamMembersScope)
 
-	// Create another test user - we will invite this one to join the team
-	invitee := g.CreateUser()
-	id := invitee.IdentityID()
+		team.AssignRole(teamAdmin.Identity(), r.Role())
 
-	invitations := []invitation.Invitation{
-		{
-			IdentityID: &id,
-			Member:     true,
-		},
-	}
+		// Create another test user - we will invite this one to join the team
+		invitee := g.CreateUser()
+		id := invitee.IdentityID()
 
-	err := s.Application.InvitationService().Issue(s.Ctx, teamAdmin.IdentityID(), team.TeamID().String(), invitations)
-	require.NoError(s.T(), err, "Error creating invitations")
+		invitations := []invitation.Invitation{
+			{
+				IdentityID: &id,
+				Member:     true,
+			},
+		}
 
-	invs, err := s.invitationRepo.ListForIdentity(s.Ctx, team.TeamID())
-	require.NoError(s.T(), err, "Error listing invitations")
+		var messages []notification.Message
+		*s.notificationServiceMock = *testservice.NewNotificationServiceMock(s.T())
+		s.notificationServiceMock.SendMessagesAsyncFunc = func(p context.Context, p1 []notification.Message) (r2 error) {
+			messages = p1
+			return nil
+		}
 
-	require.Equal(s.T(), 1, len(invs))
-	require.True(s.T(), invs[0].Member)
-}
+		err := s.Application.InvitationService().Issue(s.Ctx, teamAdmin.IdentityID(), team.TeamID().String(), invitations)
+		require.NoError(s.T(), err, "Error creating invitations")
```

Comment:
you must use `t` instead of `s.T()` in all assertions in subtests, otherwise the test will `panic` if a result is not as expected/required.